### ISSUE TITLE
Vulkan SC

### ISSUE THREAD

**Issue by tlf30** [LWJGL/lwjgl3#874]
### Library

Vulkan SC

### Project page

https://www.khronos.org/vulkansc/

### Source repository

https://github.com/KhronosGroup/VulkanSC-Docs

### Platforms

Windows x64

### Description

Hello @Spasi,
I have been roped into a project that needs (wants) to run Vulkan SC inside of Java. 
I know you are going to be changing gears to LWJGL 4 now that the latest LWJGL 3.3.2 is out, but I was wondering if you would consider adding VulkanSC bindings to LWJGL 3? (Or even LWJGL 4 if you are no longer going to do any releases to 3)
Until then we will probably move forward with regular vulkan for prototyping, but limiting our functions to be able to smoothly transition to SC. 

Thanks,
Trevor

**Comment by Spasi** [comment]
Hey @tlf30,

Supporting SC as completely separate bindings would be too much I think. Would you be OK with simply exposing whatever functions/tokens SC introduces over standard Vulkan? Afaict, most of everything else remains the same (except the unavailable functionality).

**Comment by ws909** [comment]
@Spasi Would that be introduced in a manner which makes the border between normal and SC very blurry? For instance, while writing code using normal Vulkan, the IDE would suggest SC tokens unavailable in normal Vulkan? This could be rather confusing.

**Comment by Spasi** [comment]
@ws909 Yes.

**Comment by ws909** [comment]
@Spasi Then I am not in favour of this approach. 

Another module isn't needed, but I prefer that the existing VK classes are not polluted. Can the new tokens be added to a VKSC class hierarchy? The lack of multi-inheritance requires these classes not to inherit from VK10.

**Comment by tlf30** [comment]
I also agree. While they could be the same module, they would at least need to be completely separate classpaths with crossover.
There is a lot of functionality in regular vulkan that is not available in SC, and SC features not in regular. It would be very confusing if SC classes were mixed with regular classes. Complete seperation would be a requirement, even if in the same module.
